src/attributes.md

```diff
@@ -0,0 +1,49 @@
+# Attributes
+
+Attributes come in two types - 'builtin'/'inert' attributes, and 'non-builtin'/'active' attributes:
+
+## 'Builtin'/'inert' attributes
+
+These attributes are defined in the compiler itself, in
+[`compiler/rustc_feature/src/builtin_attrs.rs`][builtin_attrs]
+
+Example include `#[allow]` and `#[macro_use]`
+
+[builtin_attrs]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_feature/builtin_attrs/index.html
+
+These attributes have several important characteristics:
+* They are always in scope, and do not participate in typical path-based resolution.
+* They cannot be renamed. For example, `use allow as foo` will compile, but writing `#[foo]` will
+  produce an error.
+* They are 'inert', meaning they are left as-is by the macro expansion code.
+  As a result, any behavior comes as a result of the compiler explicitly checking for their presence
+  (for example, lint-related code explicitly checks for `#[allow]`, `#[warn]`, `#[deny]`, and
+`#[forbid]`)
```

Comment:
Might be nice to add a bit more to make it clear what the distinction is:

```suggestion
  As a result, any behavior comes as a result of the compiler explicitly checking for their presence.
  For example, lint-related code explicitly checks for `#[allow]`, `#[warn]`, `#[deny]`, and
  `#[forbid]`, rather than the behavior coming from the expansion of the attributes themselves.
```